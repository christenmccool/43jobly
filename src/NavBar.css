.NavBar {
  display: flex;
  justify-content: space-between;
  margin: 0;
  background-color: white;
  padding: 20px;
  border-bottom: 1px solid grey;
}

.NavBar-right {
  display: flex;
  gap: 20px;
}

nav a {
  text-decoration: none;
  color: blue;
  font-size: 1em;
}

.NavBar-logout-btn {
  color: blue;
  cursor: pointer;
  background-color: white;
  border: none;
  font-size: 1em;
}

nav a.active{
  font-weight: bold;
  color: black;
}

