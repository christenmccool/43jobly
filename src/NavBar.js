import React, {useContext} from 'react';
import {NavLink} from 'react-router-dom';
import './NavBar.css';
import UserContext from './UserContext';

/** NavBar component for Jobly app
 * Display Login and Signup links when no user logged in
 * Display Companies, Jobs, Profile, and Log out link when user logged in
 */
 
const NavBar = ({logout}) => {
  const user = useContext(UserContext);

  if (user) {
    return (
      <nav className="NavBar">
        <div className="NavBar-left"> 
            <NavLink exact to="/">Jobly</NavLink>
        </div> 
        <div className="NavBar-right"> 
          <NavLink exact to="/companies">Companies</NavLink>
          <NavLink exact to="/jobs">Jobs</NavLink>
          <NavLink exact to="/applications">Applications</NavLink>
          <NavLink exact to="/profile">Profile</NavLink>
          <button onClick={logout} className="NavBar-logout-btn">Log out {user.username}</button>
        </div> 
      </nav>
    )
  } else {
    return (
      <nav className="NavBar">
        <div className="NavBar-left"> 
            <NavLink exact to="/">Jobly</NavLink>
        </div> 
        <div className="NavBar-right"> 
          <NavLink exact to="/login">Login</NavLink>
          <NavLink exact to="/signup">Signup</NavLink>
        </div> 
      </nav>
    )
  }
}

export default NavBar;