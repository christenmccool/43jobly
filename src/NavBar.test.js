import { render, asFragment, fireEvent, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NavBar from './App';
import axiosMock from "axios";

// jest.mock('axios');

test('if shows companies, jobs, and profile links when user logged in', async () => {
  const {getByText, queryByLabelText, debug} = render(<MemoryRouter><NavBar /></MemoryRouter>);

  // axiosMock.post.mockResolvedValueOnce({ data: {token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImNocmlzdGVuIiwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTYzNDQ3Njg5MX0.fDC8dNCeY3LwKh_SThuJKJ_5soUvjItHM7k6TC4roUU"}});


  //Log in as test user
  const loginLink = getByText("Login");
  fireEvent.click(loginLink);
  const username = queryByLabelText(`Username`);
  const password = queryByLabelText(`Password`);
  const loginBtn = getByText("Submit");
  fireEvent.change(username, {target: {value: "testuser"}});
  fireEvent.change(password, {target: {value: "password"}});
  fireEvent.click(loginBtn);

  await waitFor(() => expect(getByText(`Log out`, {exact: false})).toBeInTheDocument());
  debug();
  // expect(getByText(`Companies`)).toBeInTheDocument();
  // expect(getByText(`Jobs`)).toBeInTheDocument();
});





test('if renders without crashing', () => {
  render(<MemoryRouter><NavBar /></MemoryRouter>);
});

test('if matches snapshot', () => {
  const {asFragment} = render(<MemoryRouter><NavBar /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});

test('if shows jobly, login, and signup links when no user logged in', () => {
  const {getByText, getAllByText, debug} = render(<MemoryRouter><NavBar /></MemoryRouter>);

  expect(getAllByText('Jobly')).toHaveLength(2);
  expect(getByText('Login')).toBeInTheDocument();
  expect(getByText('Signup')).toBeInTheDocument();
});

test('if login and signup links work', async () => {
  const {getByText, queryByLabelText, debug} = render(<MemoryRouter><NavBar /></MemoryRouter>);
  
  expect(queryByLabelText(`Username`)).not.toBeInTheDocument();
  expect(queryByLabelText(`Password`)).not.toBeInTheDocument();
  expect(queryByLabelText(`Email`)).not.toBeInTheDocument();

  const loginLink = getByText("Login");
  fireEvent.click(loginLink);
  expect(queryByLabelText(`Username`)).toBeInTheDocument();
  expect(queryByLabelText(`Password`)).toBeInTheDocument();
  expect(queryByLabelText(`Email`)).not.toBeInTheDocument();

  const signupLink = getByText("Signup");
  fireEvent.click(signupLink);
  expect(queryByLabelText(`Username`)).toBeInTheDocument();
  expect(queryByLabelText(`Password`)).toBeInTheDocument();
  expect(queryByLabelText(`Email`)).toBeInTheDocument();
});


// test('if company and job links work', async () => {
//   const {queryByText, queryByLabelText, debug} = render(<MemoryRouter><NavBar /></MemoryRouter>);

//   //Log in as test user
//   const loginLink = queryByText("Login");
//   fireEvent.click(loginLink);
//   const username = queryByLabelText(`Username`);
//   const password = queryByLabelText(`Password`);
//   const loginBtn = queryByText("Submit");
//   fireEvent.change(username, {target: {value: "testuser"}});
//   fireEvent.change(password, {target: {value: "password"}});
//   fireEvent.click(loginBtn);
//   await waitFor(() => expect(queryByText(`Log out testuser`)).toBeInTheDocument());

//   expect(queryByText(`Companies`)).toBeInTheDocument();
//   expect(queryByText(`Jobs`)).toBeInTheDocument();
  
//   expect(queryByText(`Anderson, Arias and Morrow`)).not.toBeInTheDocument();
//   expect(queryByText(`Accommodation manager`)).not.toBeInTheDocument();

//   const companiesLink = queryByText("Companies");
//   fireEvent.click(companiesLink);
//   await waitFor(() => expect(queryByText(`Anderson, Arias and Morrow`)).toBeInTheDocument());

//   const jobsLink = queryByText("Jobs");
//   fireEvent.click(jobsLink);
//   await waitFor(() => expect(queryByText(`Accommodation manager`)).toBeInTheDocument());
// });
