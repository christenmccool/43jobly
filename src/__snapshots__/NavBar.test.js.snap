// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="NavBar"
    >
      <div
        class="NavBar-left"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Jobly
        </a>
      </div>
      <div
        class="NavBar-right"
      >
        <a
          href="/login"
        >
          Login
        </a>
        <a
          href="/signup"
        >
          Signup
        </a>
      </div>
    </nav>
    <div
      class="Routes"
    >
      <div
        class="Home"
      >
        <h1>
          Jobly
        </h1>
        <p>
          All the jobs in one convenient place.
        </p>
        <div>
          <a
            href="/login"
          >
            Log in
          </a>
          <a
            href="/signup"
          >
            Sign up
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
